homeassistant/components/history/__init__.py

```diff
@@ -116,10 +116,18 @@ async def get(
                 return self.json_message("Invalid end_time", HTTPStatus.BAD_REQUEST)
         else:
             end_time = start_time + one_day
+
+        hass = request.app["hass"]
+
         entity_ids_str = request.query.get("filter_entity_id")
         entity_ids = None
         if entity_ids_str:
             entity_ids = entity_ids_str.lower().split(",")
+            for entity_id in entity_ids:
```

Comment:
What about history for deleted entities ?

Reply:
That's a good point, I hadn't thought of that, I have added a regex check for the REST API and WS API that runs if the entity does not exist.